fprime-util new --project error in folder without permissions

| | |
|:---|:---|
|**_fprime-tools Version_**| 3.2.1|
|**_Affected Component_**|   |
---
## Problem Description

When trying to create a new project inside a directory without the right permission, a general error is returned.

## How to Reproduce

1.  Create a directory
2. Remove the permission to create file in the directory
3. Run `fprime-util new --project` inside the directory created
4. You should have the error
`[ERROR] [Errno 2] No such file or directory: '/your_folder/new_fprime_project/project.cmake`

## Expected Behavior

I would expect an error more related to the problem. Something like 

`[ERROR] Cannot created a directory, permission denied`

--- a/src/fprime/util/cookiecutter_wrapper.py
+++ b/src/fprime/util/cookiecutter_wrapper.py
@@ -212,6 +212,12 @@ def new_component(build: Build):
         print(f"{out_directory_error}", file=sys.stderr)
     except CMakeExecutionException as exc:
         print(f"[ERROR] Failed to create component. {exc}", file=sys.stderr)
+    except FileNotFoundError as e:
+        print(
+            f"{e}. Permission denied to write to the directory.",
+            file=sys.stderr,
+        )
+        return 1
     except OSError as ose:
         print(f"[ERROR] {ose}")
     return 1
@@ -257,6 +263,12 @@ def new_deployment(build: Build, parsed_args):
             file=sys.stderr,
         )
         return 1
+    except FileNotFoundError as e:
+        print(
+            f"{e}. Permission denied to write to the directory.",
+            file=sys.stderr,
+        )
+        return 1
     print(f"[INFO] New deployment successfully created: {gen_path}")
     return 0
 
@@ -285,6 +297,12 @@ def new_project(parsed_args):
     except OutputDirExistsException as out_directory_error:
         print(
             f"{out_directory_error}. Use --overwrite to overwrite (will not delete non-generated files).",
+            file=sys.stderr,
+        )
+        return 1
+    except FileNotFoundError as e:
+        print(
+            f"{e}. Permission denied to write to the directory.",
             file=sys.stderr,
         )
         return 1

